src/platform/outfits/outfits.ts

```diff
@@ -0,0 +1,103 @@
+import { Color3, WearableId } from '../../misc'
+import { JSONSchema, ValidateFunction, generateLazyValidator } from '../../validation'
+
+/**
+ * @alpha
+ */
+export type Outfit = {
+  bodyShape: WearableId
+  eyes: { color: Color3 }
+  hair: { color: Color3 }
+  skin: { color: Color3 }
+  wearables: WearableId[]
+}
+
+/**
+ * Outfit
+ * @alpha
+ */
+export namespace Outfit {
+  export const schema: JSONSchema<Outfit> = {
+    type: 'object',
+    required: ['bodyShape', 'eyes', 'hair', 'skin', 'wearables'],
+    properties: {
+      bodyShape: {
+        type: 'string'
+      },
+      eyes: {
+        type: 'object',
+        required: ['color'],
+        properties: {
+          color: Color3.schema
+        }
+      },
+      hair: {
+        type: 'object',
+        required: ['color'],
+        properties: {
+          color: Color3.schema
+        }
+      },
+      skin: {
+        type: 'object',
+        required: ['color'],
+        properties: {
+          color: Color3.schema
+        }
+      },
+      wearables: {
+        type: 'array',
+        items: {
+          type: 'string'
+        }
+      }
+    },
+    additionalProperties: true
+  }
+  export const validate: ValidateFunction<Outfit> = generateLazyValidator(schema)
+}
+
+/**
+ * @alpha
+ */
+export type Outfits = {
+  outfits: {
+    slot: number
+    outfit: Outfit
+  }[]
+  namesForExtraSlots: string[]
+}
+
+/**
+ * Avatar
+ * @alpha
+ */
+export namespace Outfits {
+  export const schema: JSONSchema<Outfits> = {
+    type: 'object',
+    required: ['outfits', 'namesForExtraSlots'],
+    properties: {
+      outfits: {
+        type: 'array',
+        items: {
+          type: 'object',
+          required: ['slot', 'outfit'],
+          properties: {
+            slot: {
```

Comment:
Perhaps make slot id unique?

Reply:
Is not as easy to set `uniqueItems = true` as the slot is within an object but not an array of slots. That's validated in `content-validator`